### DISCUSSION TITLE
How to correctly cross-compile package - `linker clang not found`

### DISCUSSION THREAD

**Discussion by qarmin** [cross-rs/cross#1451] [Q&A]
When compiling proiect with `cross build --target armv7-unknown-linux-gnueabihf --release`
and having such content in ~/.cargo/config.toml
```
[target.armv7-unknown-linux-gnueabihf]
linker = "arm-linux-gnueabihf-gcc"

[target.x86_64-unknown-linux-gnu]
linker = "clang"
rustflags = ["-C", "link-arg=-fuse-ld=/usr/bin/mold"]
```
compilation thows such errors
```
   Compiling slab v0.4.9
error: linker `clang` not found
  |
  = note: No such file or directory (os error 2)

error: could not compile `tiny-keccak` (build script) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `lock_api` (build script) due to 1 previous error
```
but when I manually remove other than arv7 linkers - content of new `~/.cargo/config.toml`
```
[target.armv7-unknown-linux-gnueabihf]
linker = "arm-linux-gnueabihf-gcc"
```
then everything works fine

Why in first situation this showed an error?

**Comment by Emilgardis** [comment]
The problem is that target.target is applied for any thing thats built using that target, which in this case includes build scripts. We dont have a good solution for this currently, see https://github.com/cross-rs/cross/pull/931
